### PR TITLE
Added transparent avatars

### PR DESCRIPTION
**Pull request by Doom032**
solves #729

Notes:

 - Because this changes the entire structure of images(allowing them to have transparency), this changes a lot of the image-related functions. This breaks images on android.
 - While refactoring to avoid code duplication, I saw opportunity to move drawimage and drawavatarimage to crossplatform code, so I did. drawimage has become draw_message_image in message.c(the only place it is used), and drawavatarimage has become draw_avatar_image in ui.c, which seemed the best place for it. The platform-defined functions for image drawing are now image_set_filter, image_set_scale and draw_image.
 - images can actually be cleaned up now (use image_free)
 - [FIXED] alpha bitmasks only use up 1 byte per pixel on Xlib, but unnecessarily use up 4 bytes per pixel on Windows. I couldn't find out how to fix that after a lot searching and trying stuff, so I found it wasn't worth it to go on with it. Could always be fixed in the future.
 - [FIXED] I found out while testing this that images with alpha look terrible at the edges on Windows. However, this is not a regression. It was already like that when avatars were introduced. There should be a way to fix this, as images use alpha in a pretty way in the drawalpha function too.

Example (on Wine)
tav2

### CODE REVIEW

**Comment by Doom032** [comment]
Hold on, I've found a way to fix Windows' ugliness and make it only need 1 byte for the alpha, but I can't seem to get the color bits right at the moment(it's pretty fiddly work). I expect I'll have this fixed some time tomorrow.

**Comment by Doom032** [comment]
@irungentoo @GrayHatter Can any of you test this on Windows and look at the draw_image function in win32/main.c? For me images with alpha(avatars) that are scaled(not 40x40 when selecting from disk) are not drawn on Windows, though they are drawn in Wine. I'm at a loss as to what would solve this, but I the AlphaBlend call seems to be failing in that draw_image function.

File: main.h [hunk 1]
```diff
@@ -219,17 +219,32 @@ void loadalpha(int bm, void *data, int width, int height);
...
+ *  notes: theoretically lowest possible scale is (1.0/65536.0), highest is 65536.0, values outside of this range will create weird issues
+ *         scaling will be rounded to pixels, so it might not be exact
  */
-void drawavatarimage(UTOX_NATIVE_IMAGE, int x, int y, int width, int height, int targetwidth, int targetheight);
-UTOX_NATIVE_IMAGE png_to_image(const UTOX_PNG_IMAGE, size_t png_size, uint16_t *w, uint16_t *h);
+void image_set_scale(UTOX_NATIVE_IMAGE *image, double scale);
+
+/* draws an utox image with or without alpha channel into the rect of (x,y,width,height) on the screen,
+ * starting at position (imgx,imgy) of the image
+ * WARNING: Windows can fail to show the image at all if the rect (imgx,imgy,width,height) contains even 1 pixel outside of
```
Comment (GrayHatter) [inline main.h]: Can you add a TODO: to this line?

**Comment by GrayHatter** [comment]
win64

looks all pretty on win8.1-x64

**Pull request merged**